File: Cc.Core/Models/OrderTable.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models;

public partial class OrderTable
{
    public long Id { get; set; }

    public long AppointmentId { get; set; }

    public double? TotalAmount { get; set; }

    public long? VatId { get; set; }

    public double? NetAmount { get; set; }

    public long? CommissionId { get; set; }

    public double? CommissionAmount { get; set; }

    public long OrderStateId { get; set; }

    public string? Notes { get; set; }

    public DateOnly? Date { get; set; }

    public DateOnly? InvoiceDate { get; set; }

    public string? InvoiceNumber { get; set; }

    public string? Taxcode { get; set; }

    public string? Accountholder { get; set; }

    public string? Vatcode { get; set; }

    public double? Deposit { get; set; }

    public double? Balance { get; set; }

    public DateTime InsertDate { get; set; }

    public string? InsertUser { get; set; }

    public DateTime UpdateDate { get; set; }

    public string? UpdateUser { get; set; }

    public virtual ICollection<Event> Event { get; set; } = new List<Event>();

    public virtual ICollection<OrderPayment> OrderPayment { get; set; } = new List<OrderPayment>();

    public virtual ICollection<OrderRow> OrderRow { get; set; } = new List<OrderRow>();

    public virtual OrderState OrderState { get; set; } = null!;
}

File: Cc.Ui/src/models/orderTable.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('orderTable')
export class OrderTable extends Entity {

    @field()
    id: number;
    @field()
    appointmentId: number;
    @field()
    totalAmount: number;
    @field()
    vatId: number;
    @field()
    netAmount: number;
    @field()
    commissionId: number;
    @field()
    commissionAmount: number;
    @field()
    orderStateId: number;
    @field()
    notes: string;
    @field()
    date: string;
    @field()
    invoiceDate: string;
    @field()
    invoiceNumber: string;
    @field()
    taxcode: string;
    @field()
    accountholder: string;
    @field()
    vatcode: string;
    @field()
    deposit: number;
    @field()
    balance: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): OrderTable {

        let item = new OrderTable();
        return item;
    }
}

@resource('orderTable')
export class OrderTableSearchModel extends Entity implements Pagination {
    
    @field()
    appointmentId?: number;
    @field()
    totalAmount?: number;
    @field()
    vatId?: number;
    @field()
    netAmount?: number;
    @field()
    commissionId?: number;
    @field()
    commissionAmount?: number;
    @field()
    orderStateId?: number;
    @field()
    notes?: string;
    @field()
    date?: string;
    @field()
    invoiceDate?: string;
    @field()
    invoiceNumber?: string;
    @field()
    taxcode?: string;
    @field()
    accountholder?: string;
    @field()
    vatcode?: string;
    @field()
    deposit?: number;
    @field()
    balance?: number;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}